src/utils/broadcast.ts

```
@@ -17,5 +24,510 @@ export function generateErrorMessage(result: BroadcastResult, errorMessage?: str
+    })
+
+  // Check for denomAmount strings like '1000uosmo' and add the coin to the return
+  event.attributes.forEach((attr: TransactionEventAttribute) => {
+    const amountDenomString = denomAmountActions.includes(attr.key) ? attr.value : undefined
+    if (amountDenomString) {
+      const coin = getCoinFromAmountDenomString(amountDenomString)
+      if (coin) coins.push({ coin: coin })
+    }
+  })
+
+  return coins
+}
+
+function getCoinFromSwapEvent(event: TransactionEvent) {
+  const tokenInAmountDenomString = event.attributes.find((a) => a.key === 'tokens_in')?.value
+  const tokenOutAmountDenomString = event.attributes.find((a) => a.key === 'tokens_out')?.value
+
+  if (!tokenInAmountDenomString || !tokenOutAmountDenomString)
+    return { tokenIn: undefined, tokenOut: undefined }
+
+  return {
+    tokenIn: getCoinFromAmountDenomString(tokenInAmountDenomString),
+    tokenOut: getCoinFromAmountDenomString(tokenOutAmountDenomString),
+  }
+}
+
+function findUpdateCoinBalanceAndAddDeposit(
+  event: TransactionEvent,
+  transactionCoins: TransactionCoin[],
+) {
+  if (event.attributes.find((a) => a.key === 'action')?.value !== 'update_coin_balance') return
+  const amountDenomString = event.attributes.find((a) => a.key === 'coin')?.value
+  if (!amountDenomString) return
+  const result = getCoinFromAmountDenomString(amountDenomString)
+  if (!result) return
+
+  const depositFromWallet = transactionCoins.find((c) => c.type === 'deposit_from_wallet')?.coin
+  return depositFromWallet ? mergeCoinAmounts(depositFromWallet, result) : result
+}
+
+function getTargetFromEvent(event: TransactionEvent, address: string): TransactionRecipient {
+  const recipient = event.attributes.find((a) => a.key === 'recipient')?.value
+
+  if (recipient && recipient === address) return 'wallet'
+  return 'contract'
+}
+
+function mergeCoinAmounts(coin1: BNCoin, coin2: BNCoin): BNCoin {
+  if (coin1.denom !== coin2.denom) return coin2
```

This seems like incorrect behaviour to me. I think it should throw an error when this happens, not the coin that you are trying to merge into the first one.

This functions responsibility is just to merge the coins. If there needs to be a denom check, it should be done outside of this function IMO.